dataflow/src/main/java/org/checkerframework/dataflow/busyexpression/BusyExprTransfer.java

```diff
@@ -0,0 +1,103 @@
+            Node n, TransferInput<UnusedAbstractValue, BusyExprStore> p) {
+        return new RegularTransferResult<>(null, p.getRegularStore());
+    }
+
+    @Override
+    public RegularTransferResult<UnusedAbstractValue, BusyExprStore> visitAssignment(
+            AssignmentNode n, TransferInput<UnusedAbstractValue, BusyExprStore> p) {
+        RegularTransferResult<UnusedAbstractValue, BusyExprStore> transferResult =
+                (RegularTransferResult<UnusedAbstractValue, BusyExprStore>)
+                        super.visitAssignment(n, p);
+        processBusyExprInAssignment(
+                n.getTarget(), n.getExpression(), transferResult.getRegularStore());
+        return transferResult;
+    }
+
+    @Override
+    public RegularTransferResult<UnusedAbstractValue, BusyExprStore> visitMethodInvocation(
+            MethodInvocationNode n, TransferInput<UnusedAbstractValue, BusyExprStore> p) {
+        RegularTransferResult<UnusedAbstractValue, BusyExprStore> transferResult =
+                (RegularTransferResult<UnusedAbstractValue, BusyExprStore>)
+                        super.visitMethodInvocation(n, p);
+        BusyExprStore store = transferResult.getRegularStore();
+        for (Node arg : n.getArguments()) {
+            store.addUseInExpression(arg);
+        }
+        return transferResult;
+    }
+
+    @Override
+    public RegularTransferResult<UnusedAbstractValue, BusyExprStore> visitObjectCreation(
+            ObjectCreationNode n, TransferInput<UnusedAbstractValue, BusyExprStore> p) {
+        RegularTransferResult<UnusedAbstractValue, BusyExprStore> transferResult =
+                (RegularTransferResult<UnusedAbstractValue, BusyExprStore>)
+                        super.visitObjectCreation(n, p);
+        BusyExprStore store = transferResult.getRegularStore();
+        for (Node arg : n.getArguments()) {
+            store.addUseInExpression(arg);
+        }
+        return transferResult;
+    }
+
+    @Override
+    public RegularTransferResult<UnusedAbstractValue, BusyExprStore> visitReturn(
+            ReturnNode n, TransferInput<UnusedAbstractValue, BusyExprStore> p) {
+        RegularTransferResult<UnusedAbstractValue, BusyExprStore> transferResult =
+                (RegularTransferResult<UnusedAbstractValue, BusyExprStore>) super.visitReturn(n, p);
+        Node result = n.getResult();
+        if (result != null) {
+            BusyExprStore store = transferResult.getRegularStore();
+            store.addUseInExpression(result);
+        }
+        return transferResult;
+    }
+
+    /**
+     * Update the information of busy expression store from an assignment statement.
+     *
+     * @param variable if any expression has this variable, that expression should be killed
+     * @param expression the expression should be added
+     * @param store the busy expression store
+     */
+    public void processBusyExprInAssignment(Node variable, Node expression, BusyExprStore store) {
```

Comment:
This function is only used once, in this class, right? Does this really improve readability? Why make the method public?

Reply:
Thanks for pointing this out! I make this private now. I deleted the `visitStringConcatenateAssignment` method which used this method. I will delete it.